src/actions/MastodonAddKernelsAction.C

```diff
@@ -0,0 +1,135 @@
+
+// MOOSE includes
+#include "Conversion.h"
+#include "Factory.h"
+#include "FEProblem.h"
+#include "MooseMesh.h"
+#include "Action.h"
+#include "MooseObjectAction.h"
+
+template <>
+InputParameters
+validParams<MastodonAddKernelsAction>()
+{
+  InputParameters params = validParams<Action>();
+  params.addClassDescription("Create inertia Kernels, and acceleration, velocity, stress and "
+                             "strain AuxKernels automatically, based on the dimension of the "
+                             "problem.");
+  params.addParam<bool>("inertia_kernels", true, "false, if inertia Kernels are created manually.");
+  params.addParam<bool>(
+      "auxkernels",
+      true,
+      "false, if velocity, acceleration, stress and strain AuxKernels are created manually.");
+  params.addParam<Real>("beta", 0.25, "beta parameter for Newmark time integration.");
+  params.addParam<Real>("gamma", 0.5, "gamma parameter for Newmark time integration.");
+  return params;
+}
+
+MastodonAddKernelsAction::MastodonAddKernelsAction(const InputParameters & params) : Action(params)
+{
+}
+
+void
+MastodonAddKernelsAction::act()
+{
+  const bool inertia_kernels = getParam<bool>("inertia_kernels");
+  const bool auxkernels = getParam<bool>("auxkernels");
+  const std::vector<VariableName> disp_variables = {"disp_x", "disp_y", "disp_z"};
+  const std::vector<AuxVariableName> vel_variables = {"vel_x", "vel_y", "vel_z"};
+  const std::vector<AuxVariableName> accel_variables = {"accel_x", "accel_y", "accel_z"};
+  const std::vector<std::string> comp = {"x", "y", "z"};
+
+  // Adding inertia kernels
+  if (inertia_kernels && _current_task == "add_kernel")
+  {
+    for (std::size_t i = 0; i < _problem->mesh().dimension(); i++)
+    {
+      InputParameters params = _factory.getValidParams("InertialForce");
+      params.set<NonlinearVariableName>("variable") = disp_variables[i];
+      params.set<std::vector<VariableName>>("velocity") = {vel_variables[i]};
+      params.set<std::vector<VariableName>>("acceleration") = {accel_variables[i]};
+      params.set<Real>("beta") = getParam<Real>("beta");
+      params.set<Real>("gamma") = getParam<Real>("gamma");
+      _problem->addKernel("InertialForce", "inertia" + comp[i], params);
+    }
+  }
+
+  // Adding AuxKernels
+  if (auxkernels && _current_task == "add_kernel")
+  {
+    // velocity and acceleration AuxKernels
+    const std::vector<std::string> vel_auxkernel = {"vel_x", "vel_y", "vel_z"};
+    const std::vector<std::string> accel_auxkernel = {"accel_x", "accel_y", "accel_z"};
+    const Real beta = getParam<Real>("beta");
+    const Real gamma = getParam<Real>("gamma");
+    for (std::size_t j = 0; j < _problem->mesh().dimension(); j++)
+    {
+      // Velocity AuxKernels: These Auxkernels are added as MooseObjectActions
+      // due to conflicts with DynamicTensorMechanicsAction when added using
+      // addAuxKernel method. TODO: Investigate this conflict. Talk to Ben Spencer.
```

Comment:
I am surprised that you are able to set up NewmarkAccelAux using addAuxKernel but not NewarkVelAux. They both derive from AuxKernel and should behave identically.

Reply:
@sveerara, @aeslaughter and I were surprised as well. It looks like the DynamicTensorMechanicsAction is somehow interfering with this and does not create the velocity AuxKernels when using NewmarkVelAux. We are still working on this and hopefully we will find the reason why.